cheatsheets/Cross-Site_Request_Forgery_Prevention_Cheat_Sheet.md

```diff
@@ -45,7 +41,7 @@ This defense is one of the most popular and recommended methods to mitigate CSRF
 
 ### Synchronizer Token Pattern
 
-Any state changing operation requires a secure random token (e.g., CSRF token) to prevent CSRF attacks. A CSRF token should be unique per user session, large random value, and also generated by a cryptographically secure random number generator. The CSRF token is added as a hidden field for forms, headers/parameters for AJAX calls (It is recommended to add in parameter than in header. If you need to add it to header, it is better to make sure that the token header is not being logged at your server) and within the URL if the state changing operation occurs via a GET. See "Disclosure of Token in URL" section below. The server rejects the requested action if the CSRF token fails validation.
+Any state changing operation requires a secure random token (e.g., CSRF token) to prevent CSRF attacks. A CSRF token should be unique per user session, a large random value, and also generated by a cryptographically secure random number generator. The CSRF token can be added through hidden fields, headers, additional parameters in forms, AJAX calls and within the URL. Token should be added to all state changing actions. Make sure that token will not leakue for example in server logs (e.g. when server log all of the headers) or [in the URL](#disclosure-of-token-in-url). The server rejects the requested action if the CSRF token fails validation.
```

Comment:
What do you think about the following suggestion where I:
- Improved the way the CSRF token is described
- Improved the ease of readability
- Removed sending the CSRF token as part of the URL. It doesn't make sense sending it in the URL and stating right after that URLs get logged and we are saying not to log it?
- Made a small separation between where the CSRF token is added, and what functions use it.
```suggestion
Any state changing operation requires a secure random token (e.g., CSRF token) to prevent CSRF attacks. A CSRF token should be:

- unique per user session;
- large random value;
- generated by a Cryptographically Secure Pseudo-Random Number Generator (CSPRNG).

The CSRF token can be added through hidden fields, headers, and can be used with forms, and AJAX calls. That token should be added to all state changing actions. Make sure that the token is not leaked in the server logs, or [in the URL](#disclosure-of-token-in-url). The server must reject the requested action if the CSRF token fails validation.
```

Reply:
Looks much better! 
I see also one one change that we can add here. In the first sentence we write "Any state changing operation requires a secure random token" and that we repet it "That token should be added to all state changing actions". Maybe we should delete this repetition?

Reply:
Certainly!